lib/pbench/server/api/resources/users_api.py

```diff
@@ -300,8 +300,52 @@ def __init__(self, logger, auth):
         self.logger = logger
         self.auth = auth
 
+    def get_valid_target_user(
+        self, target_username: str, request_type: str
+    ) -> NamedTuple:
+        """
+        Helper function to determine whether the API call is permitted for the target username
+        Right now it is only permited for an admin user and the target user itself
+        :param target_username: str
+        :param request_type: str
+        :return: NamedTuple containing target User object/ None, http_status_to_return, http_message_to_return
+        """
+        TargetUser = NamedTuple(
+            "TargetUser",
+            [("user", User), ("http_status", HTTPStatus), ("http_message", str)],
+        )
```

Comment:
Hmm; can you really define a named tuple from the `typing` module? I'm not sure that's a good idea, in any case.

Also, this should be defined at module level (or at least at class level) so it's accessible to callers. Plus the return type hint should be `TargetUser` rather than just a generic `NamedTuple`, which doesn't say much.

You can also define a named tuple with defaults for right-most fields; so, for example, you could default to `(HTTPStatus.OK, "")` and most of your returns would become `return TargetUser(user)`. Although it might be easier for readers to keep track of if you specified all three by keyword, e.g. `return TargetUser(user=name, http_status=HTTPStatus.OK, str="")` (should that be `message` instead of `str`?)

Reply:
Yes, you can do this, although the documentation for it seems to be thin on the ground.  The principal advantage of this seems to be that it allows one to specify the types of the named members.

And, yes, I too thought it might be nice to set up default values for the (right-most) members, but I couldn't figure out how to get `NamedTuple` to accept the default values.  (Having the HTTP stuff disappear in the success case would actually be a feature; having the user disappear in the failure case would not necessarily be bad, either.)

> should that be message instead of str?

Dave, I think you're reading that wrong -- I think the field names are the first elements, and their types are the second elements, so `http_message` is the name of the third member and its type is `str`.

Reply:
Yeah I used `typing` module because we can specify the types of the named members